tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/editor/inserthandlers/BallerinaEnterInDocumentationHandler.java

```diff
@@ -29,48 +30,53 @@
 import org.jetbrains.annotations.NotNull;
 
 /**
- * Handles the enter key press in braces.
+ * Handles the enter key press events within ballerina documentation.
  */
 public class BallerinaEnterInDocumentationHandler extends EnterHandlerDelegateAdapter {
 
+    private static final String BAL_DOC_PREFIX = "#";
+
     @Override
     public Result postProcessEnter(@NotNull PsiFile file, @NotNull Editor editor, @NotNull DataContext dataContext) {
-        if (!file.getLanguage().is(BallerinaLanguage.INSTANCE)) {
+
+        if (!file.getLanguage().is(BallerinaLanguage.INSTANCE) || editor.isDisposed()) {
             return Result.Continue;
         }
+        Document doc = editor.getDocument();
+
         // We need to save the file before checking. Otherwise issues can occur when we press enter in a string.
         Project project = file.getProject();
-        PsiDocumentManager.getInstance(project).commitDocument(editor.getDocument());
+        PsiDocumentManager.getInstance(project).commitDocument(doc);
 
-        // Checks whether the previous line starts with "#".
         LogicalPosition caretPos = editor.getCaretModel().getLogicalPosition();
         int prevLine = caretPos.line - 1;
-        String lineString = editor.getDocument().getText(
-                new TextRange(editor.getDocument().getLineStartOffset(prevLine),
-                        editor.getDocument().getLineEndOffset(prevLine))).trim();
-        if (lineString.startsWith("#")) {
-            int newCol = lineString.replace("\t", "    ").indexOf("#");
-            String enteredText = editor.getDocument().getText(
-                    new TextRange(editor.getDocument().getLineStartOffset(caretPos.line),
-                            editor.getDocument().getLineEndOffset(caretPos.line))).trim();
-            editor.getDocument().deleteString(editor.getDocument().getLineStartOffset(caretPos.line),
-                    editor.getDocument().getLineEndOffset(caretPos.line));
-            editor.getCaretModel().moveToLogicalPosition(new LogicalPosition(caretPos.line, 1));
-            enterNewLine(editor, enteredText, newCol);
+        String lineString = doc.getText(new TextRange(doc.getLineStartOffset(prevLine),
+                doc.getLineEndOffset(prevLine)));
 
+        if (lineString.trim().startsWith(BAL_DOC_PREFIX)) {
+            addNewline(editor, doc, lineString, caretPos);
             // Commit the document.
-            PsiDocumentManager.getInstance(project).commitDocument(editor.getDocument());
+            PsiDocumentManager.getInstance(project).commitDocument(doc);
         }
         return Result.Continue;
     }
 
-    private void enterNewLine(Editor editor, String str, int col) {
-        StringBuilder strBuilder = new StringBuilder("# " + str);
-        // Left padding with whitespaces in order to be vertically aligned with the previous doc line.
-        for (int i = 0; i < col; i++) {
+    // Inserts the splitted documentation at the next line.
```

Comment:
Better to use method documentation format instead of the single line comment

Reply:
@nadeeshaan As per my knowledge,
1. javadocs comes forward when the info presents which the user of a class needs to know and 
2. Info that a developer modifying a class might better go into a normal comment (block or line).

Above context (private method without function args descriptions) is more related to the scenario (2) IMO :) WDYT?

Please also refer this [SO Thread](https://stackoverflow.com/questions/3607641/javadoc-comments-vs-block-comments) as well.